tests/src/Integration/Condition/DataIsEmptyTest.php

```diff
@@ -39,17 +39,17 @@ public function setUp() {
    */
   public function testConditionEvaluation() {
     // Test a ComplexDataInterface object.
-    $entity_adapter_empty = $this->prophesize(ComplexDataInterface::class);
-    $entity_adapter_empty->isEmpty()->willReturn(TRUE)->shouldBeCalledTimes(1);
-
-    $this->condition->getContext('data')->setContextData($entity_adapter_empty->reveal());
-    $this->assertTrue($this->condition->evaluate());
-
-    $entity_adapter_full = $this->prophesize(ComplexDataInterface::class);
-    $entity_adapter_full->isEmpty()->willReturn(FALSE)->shouldBeCalledTimes(1);
-
-    $this->condition->getContext('data')->setContextData($entity_adapter_full->reveal());
-    $this->assertFalse($this->condition->evaluate());
+    //$entity_adapter_empty = $this->prophesize(ComplexDataInterface::class);
```

Comment:
What exactly are these mocks intended to test?

As I said before, the mocks need to change.  The following works, but I don't know if this tests what you really want tested:

```php
    // Test a ComplexDataInterface object.
    $data_definition_interface = $this->prophesize(DataDefinitionInterface::class);
    $data_definition_interface->getDataType()->willReturn('any');
    
    $entity_adapter_empty = $this->prophesize(ComplexDataInterface::class);
    $entity_adapter_empty->getDataDefinition()->willReturn($data_definition_interface);
    $entity_adapter_empty->getValue()->willReturn(FALSE);
        
    $this->condition->getContext('data')->setContextData($entity_adapter_empty->reveal());
    $this->assertTrue($this->condition->evaluate());
    
    $entity_adapter_full = $this->prophesize(ComplexDataInterface::class);
    $entity_adapter_full->getDataDefinition()->willReturn($data_definition_interface);
    $entity_adapter_full->getValue()->willReturn(TRUE);
    
    $this->condition->getContext('data')->setContextData($entity_adapter_full->reveal());
    $this->assertFalse($this->condition->evaluate());
```

`isEmpty()` has to go, because it really is never called:  by the time doEvaluate() sees the complex data, it's already been unwrapped to the data itself.  The call to `getValue()` has already done the conversion.

What do you you guys want me to do here?